packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -448,6 +466,91 @@ void main() {
     expect(find.text('B'), findsOneWidget);
   });
 
+  testWidgets('replaceNamed triggers secondaryAnimation', (WidgetTester tester) async {
+    final Map<String, WidgetBuilder> routes = <String, WidgetBuilder>{
+      '/' : (BuildContext context) => OnTapPage(id: '/', onTap: () { Navigator.pushReplacementNamed(context, '/A'); }),
+      '/A': (BuildContext context) => OnTapPage(id: 'A', onTap: () { Navigator.pushReplacementNamed(context, '/B'); }),
+      '/B': (BuildContext context) => const OnTapPage(id: 'B'),
+    };
+
+    await tester.pumpWidget(MaterialApp(
+      onGenerateRoute: (RouteSettings settings) {
+        return SlideInOutPageRoute<dynamic>(bodyBuilder: routes[settings.name]);
+      }));
```

Comment:
Nit: closing punctuation alignment should match opening - [link](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#if-you-have-a-newline-after-some-opening-punctuation-match-it-on-the-closing-punctuation)